systemlink-notebook-datasource/src/QueryEditor.tsx

```diff
@@ -70,24 +68,19 @@ export class QueryEditor extends PureComponent<
   };
 
   onParameterChange = (id: string, value: string) => {
-    const { onChange, onRunQuery } = this.props;
-    const query = defaults(this.props.query, defaultQuery);
-    const parameters = query.parameters;
-    parameters[id] = this.formatParameterValue(id, value);
-    onChange({ ...query, parameters });
+    const { onChange, onRunQuery, query } = this.props;
+    onChange({ ...query, parameters: { ...query.parameters, [id]: this.formatParameterValue(id, value) } });
```

Comment:
Can we improve the readability of this?

Reply:
Readability is in the eye of the beholder 😋
Joking aside, this pattern of spread operators to do immutable updates is pretty common in React: https://stackoverflow.com/questions/43040721/how-to-update-nested-state-properties-in-react
One of the answers mentioned a package 'immutability-helper' to make this cleaner, so that may be worth testing out another time.
I went ahead and made a small change to pull out the formatted value into a variable to make it a little easier to read.